libraries/src/Console/UpdateCoreCommand.php

```diff
@@ -386,4 +401,34 @@ public function copyFileTo($file, $dir): void
     {
         Folder::copy($file, $dir, '', true);
     }
+
+    /**
+     * Check the schema version
+     *
+     * @return  integer the number of errors
+     *
+     * @since __DEPLOY_VERSION__
+     */
+    public function checkSchema(): int
+    {
+        $app       = $this->getApplication();
+        $app->getLanguage()->load('com_installer', JPATH_ADMINISTRATOR);
+        $errors  = 0;
+        $coreExtensionInfo = ExtensionHelper::getExtensionRecord('joomla', 'file');
+
+        $dbmodel = $app->bootComponent('com_installer')->getMVCFactory($app)->createModel('Database', 'Administrator');
+
+        // Ensure we only get information for core
+        $dbmodel->setState('filter.extension_id', $coreExtensionInfo->extension_id);
+        $changeSet = $dbmodel->getItems();
```

Comment:
```suggestion
        $changeSet = $dbmodel->getItems()[0];
```
See https://github.com/joomla/joomla-cms/pull/40468#issuecomment-1519847613 .

Reply:
it's already setted in the model state  with `$coreExtensionInfo->extension_id`

Reply:
And later below no foreach.

Reply:
Let me fix my proposed code change so it covers all. I will close this one here and make a new one in a few minutes.